content/en/tracing/manual_instrumentation/cpp.md

```diff
@@ -1,23 +1,68 @@
 ---
-title: C++ Manual Instrumentation
+title: C++ Custom Instrumentation
 kind: documentation
 decription: 'Manually instrument your C++ application to send custom traces to Datadog.'
 further_reading:
-    - link: 'tracing/guide/instrument_custom_method'
-      tag: 'Guide'
-      text: 'Instrument a custom method to get deep visibility into your business logic'
     - link: 'tracing/connect_logs_and_traces'
       tag: 'Documentation'
       text: 'Connect your Logs and Traces together'
-    - link: 'tracing/opentracing'
-      tag: 'Documentation'
-      text: 'Implement Opentracing across your applications'
     - link: 'tracing/visualization/'
       tag: 'Documentation'
       text: 'Explore your services, resources, and traces'
 ---
 
-To manually instrument your code, install the tracer as in the setup examples, and then use the tracer object to create [spans][1].
+## Add tags
+
+Add custom [span tags][1] to your [spans][2] to customize your observability within Datadog.  The span tags are applied to your incoming traces, allowing you to correlate observed behavior with code-level information such as merchant tier, checkout amount, or user ID.
+
+C++ tracing uses "common tags".  These tags can be sourced from both [Datadog specific tags][3] or [OpenTracing tags][4], and included via the below:
+
+```cpp
+#include <opentracing/ext/tags.h>
+#include <datadog/tags.h>
+```
+
+Note that the Datadog tags are necessary for [unified service tagging][5].
+
+### Add custom span tags
+
+Add [tags][1] directly to a [span][2] object by calling `Span::SetTag`. For example:
+
+```cpp
+auto tracer = ...
+auto span = tracer->StartSpan("operation_name");
+span->SetTag("key must be string", "Values are variable types");
+span->SetTag("key must be string", 1234);
+```
+
+Values are of [variable type][6] and can be complex objects. Values are serialized as JSON, with the exception of a string value being serialized bare (without extra quotation marks).
+
+### Adding tags globally to all spans
```

Comment:
This section should probably reference the `DD_TAGS` environment variable instead.

Reply:
Thanks - updated!